### ISSUE TITLE
Invalid GRPC Argument when using MatchingEngine + Streaming Index

### ISSUE THREAD

**Issue by chrisle** [hwchase17/langchainjs#1612]
### Issue
I created a streaming index then tried to instantiate a class using `MatchingEngine.from_components`. When I do I get an error deeper down from `aiplatform.MatchingEngineIndex` that says the GRPC call has an invalid argument. 

```
  File "/home/jupyter/path/to/my/code/.venv/lib/python3.10/site-packages/google/api_core/grpc_helpers.py", line 74, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.InvalidArgument: 400 Request contains an invalid argument.
```

* Does`MatchingEngine.from_components` support streaming indexes ([official docs](https://cloud.google.com/vertex-ai/docs/matching-engine/create-manage-index#create-stream)) or only batch indexes?
* Is this problem at the `aiplatform.MatchingEngineIndex` level as opposed to `langchain.vectorstores.MatchingEngine`?
* Does a streaming index require a different set of arguments that's not given to `import aiplatform` when making the GRPC call  which may explain why `return aiplatform.MatchingEngineIndex` would fail?

---
### Additional Details

cc @afirstenberg who I believe is working on this section of code.
cc @c0dyhi11 

Tracing the methods:
* [Code](https://github.com/hwchase17/langchain/blob/20e9ce8a62a0747dcfb6494eddf5eb10760b9982/langchain/vectorstores/matching_engine.py#L245)`MatchingEngine.from_components`
* [Code](https://github.com/hwchase17/langchain/blob/20e9ce8a62a0747dcfb6494eddf5eb10760b9982/langchain/vectorstores/matching_engine.py#LL276C38-L276C38)`index = cls._create_index_by_id(index_id, project_id, region, credentials)` 
* [Code](https://github.com/hwchase17/langchain/blob/20e9ce8a62a0747dcfb6494eddf5eb10760b9982/langchain/vectorstores/matching_engine.py#L382)`return aiplatform.MatchingEngineIndex` 
* [Code](https://github.com/googleapis/python-aiplatform/blob/69c5f60bfea5308589a58c2b9ad9f392b3a1283e/google/cloud/aiplatform/matching_engine/matching_engine_index.py#L89)`self._gca_resource = self._get_gca_resource(resource_name=index_name)` 

Which fails:

```
  File "/home/jupyter/path/to/my/code/.venv/lib/python3.10/site-packages/google/api_core/grpc_helpers.py", line 74, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.InvalidArgument: 400 Request contains an invalid argument.
```

---

### Recreate issue

* Python 3.10.10
* Debian 11 + Pytorch on Vertex AI Workbench (it also happens on MacOS)
* langchain 0.0.197

I believe you can recreate the issue like this.

#### 1. Create a streaming index and deploy it:

```bash
curl -X POST -H "Content-Type: application/json" \
  -H "Authorization: Bearer `gcloud auth print-access-token`" \
  ${ENDPOINT}/v1/projects/${PROJECT_ID}/locations/${REGION}/indexes \
  -d '{
      displayName: "'${DISPLAY_NAME}'",
      description: "'${DISPLAY_NAME}'",
      metadata: {
        contentsDeltaUri: "'${BUCKET_NAME}'",
        config: {
            dimensions: "'${DIMENSIONS}'",
            approximateNeighborsCount: 150,
            distanceMeasureType: "DOT_PRODUCT_DISTANCE",
            algorithmConfig: {treeAhConfig: {leafNodeEmbeddingCount: 10000, leafNodesToSearchPercent: 2}}
        },
      },
      indexUpdateMethod: "STREAM_UPDATE"
  }'
```

#### 2. Instantiate your vector store like this
```python
vector_store = MatchingEngine.from_components(
    project_id="my-project",
    region="us-central1",
    gcs_bucket_name="my-bucket",
    index_id="my-index",
    endpoint_id="my_index_endpoint"
)
print("OK!")
```

**Expect result:** 🥳

**Actual result:** 😢

It never reaches `print("OK!")`

```
  File "/home/jupyter/path/to/my/code/.venv/lib/python3.10/site-packages/google/api_core/grpc_helpers.py", line 74, in error_remapped_callable
    raise exceptions.from_grpc_error(exc) from exc
google.api_core.exceptions.InvalidArgument: 400 Request contains an invalid argument.
```

**Comment by afirstenberg** [comment]
Appreciate the mention to make me aware of this, but I'm a little confused. This is opened against langchainjs, but references the Python code and the langchain library?

**Comment by chrisle** [comment]
@afirstenberg Over the weekend, I managed to answer my own question.

TLDR: When using `MatchingEngine.from_components` it assumes that you're pointing it to a Matching Engine created in batch mode. It doesn't appear to work if the index was set to  stream mode.

So I'm working on my own implementation that works with indexes created in stream mode.
